src/passes/hash-stringify-walker.cpp

```diff
@@ -68,4 +68,33 @@ void HashStringifyWalker::visitExpression(Expression* curr) {
   }
 }
 
+std::vector<SuffixTree::RepeatedSubstring> StringifyProcessor::dedupe(
+  const std::vector<SuffixTree::RepeatedSubstring> substrings) {
+  std::set<uint32_t> seen;
```

Comment:
```suggestion
  std::unordered_set<uint32_t> seen;
```
Using an unordered_set (i.e. a hash set) should be faster when iteration order doesn't matter.

Reply:
done